docs/site/migration/models/mixins.md

```diff
@@ -6,8 +6,699 @@ sidebar: lb4_sidebar
+- generating a repository and controller using the CLI as usual
+- adjusting the repository and controller files to make use of its respective
+  mixin class factory function
+
+### Defining A Common Interface For The findByTitle Method
+
+Let's define a common interface `FindByTitleInterface` in
+**src/mixins/findByTitleInterface.ts**.
+
+{% include code-caption.html content="src/mixins/findByTitleInterface.ts" %}
+
+```ts
+import {Model} from '@loopback/repository';
+
+export interface FindByTitleInterface<M extends Model> {
+  findByTitle(title: string): Promise<M[]>;
+}
+```
+
+### Defining A Repository Mixin Class Factory Function
+
+In **src/mixins/findByTitleRepositoryMixin.ts**, let's define the mixin class
+factory function `FindByTitleRepositoryMixin` which adds the `findByTitle`
+method to any repository.
+
+{% include code-caption.html content="src/mixins/findByTitleRepositoryMixin.ts" %}
+
+```ts
+import {Constructor} from '@loopback/context';
+
+import {Model} from '@loopback/repository';
+import {FindByTitleInterface} from './findByTitleInterface';
+
+/*
+ * This function adds a new method 'findByTitle' to a repository class
+ * where 'M' is a model and 'R' is DefaultCrudRepository
+ */
+export function FindByTitleRepositoryMixin<
+  M extends Model,
+  R extends Constructor<any>
+>(superClass: R) {
+  return class extends superClass implements FindByTitleInterface<M> {
+    async findByTitle(title: string): Promise<M[]> {
+      const titleFilter = {
+        where: {
+          title: title,
+        },
+      };
+      return await this.find(titleFilter);
+    }
+  };
+}
+```
+
+### Generating A Repository Via The CLI
+
+A CLI-generated repository for a model `Note` would look like this:
+
+{% include code-caption.html content="src/repositories/note.repository.ts" %}
+
+```ts
+export class NoteRepository extends DefaultCrudRepository<
+  Note,
+  typeof Note.prototype.id,
+  NoteRelations
+> {
+  constructor(@inject('datasources.db') dataSource: DbDataSource) {
+    super(Note, dataSource);
+  }
+}
+```
+
+### Adjust NoteRepository To Use FindByTitleRepositoryMixin
+
+The repository file only requires a few adjustments:
+
+- import the `FindByTitleRepositoryMixin` mixin class factory function
+- adjust the declaration of the `NoteRepository` class to extend the class
+  returned from the mixin function which takes in the `DefaultCrudRepository`
+  superclass as input.
+
+{% include code-caption.html content="src/repositories/note.repository.ts" %}
+
+```ts
+import {FindByTitleRepositoryMixin} from '../mixins/findByTitleRepositoryMixin';
+import {DefaultCrudRepository} from '@loopback/repository';
+import {Note, NoteRelations} from '../models';
+import {DbDataSource} from '../datasources';
+import {inject, Constructor} from '@loopback/core';
+
+export class NoteRepository extends FindByTitleRepositoryMixin<
+  Note,
+  Constructor<
+    DefaultCrudRepository<Note, typeof Note.prototype.id, NoteRelations>
```

Comment:
Can it be as simple as `Constructor<Note>` or `typeof Note`?

Reply:
> Can it be as simple as `Constructor<Note>` or `typeof Note`?

Can you give an example of what you mean? thx. :)

Reply:
To replace `DefaultCrudRepository<Note, typeof Note.prototype.id, NoteRelations>`

Reply:
@raymondfeng , I cannot get this to work in the suggested manner.